src/PolyType/Abstractions/IEnumTypeShape.cs

```diff
@@ -20,9 +20,20 @@ public interface IEnumTypeShape : ITypeShape
 [InternalImplementationsOnly]
 public interface IEnumTypeShape<TEnum, TUnderlying> : ITypeShape<TEnum>, IEnumTypeShape
     where TEnum : struct, Enum
+    where TUnderlying : unmanaged
```

Comment:
Seems like a reasonable constraint to add, although I'm curious what motivated its inclusion.

Reply:
I figured you might be. :)
It was required as part of the acceptance test. The CborSerializer needed a reverse-lookup table, which meant defining a `Dictionary<TUnderlying, string>` type. But `Dictionary` wants `notnull` applied to its keys. Rather than constrain `TUnderlying` merely to `notnull`, I first used `struct`, then realized `unmanaged` is the most constraining I could do, and yet guaranteed to not be overly-constraining.